src/Model/Types.ts

```diff
@@ -20,10 +20,14 @@ export type NavigationAction = RNNavigationAction & {
     name?: string,
     params?: Record<string, unknown>,
   },
+  // requireConditions
+  conditionList?: Condition[],
   // navigate
   flow?: boolean,
   reset?: boolean,
   skipConditionalNavigation?: boolean,
+  // back
+  count?: number,
```

Comment:
let's type for navigation action so we accidentally won't use parameters that are not related to particular type

```
export type NavigationAction = {
   type: 'back'
   count?: number
} | {
   type: 'navigate'
   flow?: boolean
} 
```